docs/design-guides/endpoint-and-responses.md

```diff
@@ -0,0 +1,205 @@
+
+            ```json
+            {
+                "message": "Successfully deleted the ACS user identity <communication-services-identity> which revokes all active access tokens and removes all the persisted content, and the identity mapping"
+            }
+            ```
+
+         2. If not, then return an error message
+   
+            ```json
+            {
+                "code": 500,
+                "message": "<message>",
+                "stack_trace": "<stack_trace>"
+            }
+            ```
+
+      2. If not, then return an error message
+
+         ```json
+         {
+             "code": 500,
+             "message": "An error occured when deleting the identity mapping information: <error_message>"
+         }
+         ```
+   
+   2. If not, then return an error message
+   
+      ```json
+      {
+          "code": 500,
+          "message": "<message>",
+          "stack_trace": "<stack_trace>"
+      }
+      ```
+
+## Token Endpoint
+
+The `/token` endpoint only consists of one operation - `GET` used to **get** and **refresh** Communication Services tokens
+
+### ***GET*** /token
+
+When calling the endpoint, the first step is to check if there is an existing identity mapping information stored in Microsoft Graph. If not, a new Communication Services identity will be created and stored by this endpoint.
+
+![GET Token Endpoint Sequence Diagram](../images/get-token-endpoint-sequence-diagram.png)
+
+ Detailed process shows as follow:
+
+1. Get an Azure AD token exchanged via OBO flow. If failing to get an exchanged AAD token through the request *authorization* header, then return an error message.
+
+2. Retrieve the Communication Services identity from Microsoft Graph.
+
+   1. If the identity mapping information exists, create a Communication Services token using `getToken`.
+
+      ```json
+      {
+          "token": "<communication_services_token>",
+          "expiresOn": "<token_expire_time_like_2022-01-10T00:48:44.507Z>"
+      }
+      ```
+   
+   2. If no identity mapping information exists, create a Communication Services identity and token `createUserAndToken` first.
+   
+      1. If failing to create an ACS identity and token, then return an error message.
+   
+         ```json
+         {
+             "code": 500,
+             "message": "<message>",
+             "stack_trace": "<stack_trace>"
+         }
+         ```
+   
+      2. If successful to create an ACS identity and token, then add the identity mapping information to the user resource using Graph open extensions.
+      
+         1. If successful to add, return the Azure Communication Services token to users.
+   
+            ```json
+            {
+                "token": "<communication_services_token>",
+                "expiresOn": "<token_expire_time_like_2022-01-10T00:48:44.507Z>"
+            }
+            ```
+         
+         2. If not, then return an error message.
+         
+            ```json
+            {
```

Comment:
Some of them are not latest.

![image](https://user-images.githubusercontent.com/86329602/152037657-d6866399-0d07-4380-91f4-28ead33fc2f1.png)